src/main/java/org/codeforamerica/messaging/services/MessageService.java

```diff
@@ -1,55 +1,84 @@
 package org.codeforamerica.messaging.services;
 
 import lombok.extern.slf4j.Slf4j;
-import org.codeforamerica.messaging.models.EmailMessage;
-import org.codeforamerica.messaging.models.Message;
-import org.codeforamerica.messaging.models.MessageRequest;
-import org.codeforamerica.messaging.models.SmsMessage;
+import org.codeforamerica.messaging.models.*;
 import org.codeforamerica.messaging.repositories.MessageRepository;
+import org.codeforamerica.messaging.repositories.TemplateRepository;
 import org.springframework.stereotype.Service;
 
 import java.util.Optional;
 
+import static org.codeforamerica.messaging.models.Template.DEFAULT_LANGUAGE;
+import static org.codeforamerica.messaging.models.Template.DEFAULT_VARIANT;
+
 
 @Service
 @Slf4j
 public class MessageService {
     private final SmsService smsService;
     private final EmailService emailService;
     private final MessageRepository messageRepository;
+    private final TemplateRepository templateRepository;
 
 
-    public MessageService(SmsService smsService, EmailService emailService, MessageRepository messageRepository) {
+    public MessageService(SmsService smsService,
+            EmailService emailService,
+            MessageRepository messageRepository,
+            TemplateRepository templateRepository) {
         this.smsService = smsService;
         this.emailService = emailService;
         this.messageRepository = messageRepository;
+        this.templateRepository = templateRepository;
     }
 
     public Message sendMessage(MessageRequest messageRequest) {
+        String language = getLanguage(messageRequest);
+        String variant = getVariant(messageRequest);
+        Template template = templateRepository.findFirstByNameAndLanguageAndVariant(
+                messageRequest.getTemplateName(), language, variant);
```

Comment:
@vrajmohan Specifically here I have updated the query to find the template to use name+language+variant using either default values of "en"+"A" or values passed in with `templateParams`. In the future when we have batches, the name of the template would be stored at `MessageBatch` level, and each `MessageRequest` can identify the specific `Template`. Does this resolve your question about foreign keys?

Reply:
I have also added a story to choose a variant programmatically instead of using a parameter
https://www.pivotaltracker.com/story/show/185162330

Reply:
No, it doesn't resolve my question about foreign keys.
> A foreign key must reference columns that either are a primary key or form a unique constraint.

The name of the template _by itself_ is neither a primary key nor does it form a unique constraint on the `template` table. With this design, we are precluded from enforcing referential integrity at the database level.

Reply:
We would like to declare referential integrity in a manner similar to https://github.com/codeforamerica/messaging/blob/main/src/main/resources/db/migration/V3__Create_message_table.sql#L8-L9.